coordinator/manifest/manifest.go

```diff
@@ -457,14 +461,37 @@ func EncodeSecretDataToBase64(data interface{}) (string, error) {
 	return base64.StdEncoding.EncodeToString([]byte(raw)), nil
 }
 
-// ManifestTemplateFuncMap defines the functions which can be specified for secrets in the in go template format
-var ManifestTemplateFuncMap = template.FuncMap{
+// EncodeEnvSecretDataToRaw encodes only secrets of type plain to raw byte string
+func EncodeEnvSecretDataToRaw(data interface{}) (string, error) {
+	switch secret := data.(type) {
+	case Secret:
+		if secret.Type != "plain" {
+			return "", errors.New("only secrets of type plain are allowed to use raw encoding for environment variables")
```

Comment:
I would still check if these secrets do contain unexpected NULL bytes and also throw an error in this case.

Reply:
I have added a general check for NULL bytes during `SetManifest` and moved the check at runtime during a Marble activation so it catches all NULL bytes.

I think this is better than to have the check in the template function itself, since NULL bytes can also occur if we dont use templates at all. For example through `base64` encoded env variables.

Reply:
I think checks in `SetManifest` and `WriteSecrets` for user-defined secrets are sufficient, does not necessarily be in the template function if these are the only two functions deploying new secrets.

An extra error during Marble activation is fine I guess, though not necessary cause NULL bytes are actually catched by `os.Setenv`. It's just the error is ugly and non-explanatory by default (it's just "invalid argument").

Reply:
> I think checks in `SetManifest` and `WriteSecrets` for user-defined secrets are sufficient, does not necessarily be in the template function if these are the only two functions deploying new secrets.
> 
> An extra error during Marble activation is fine I guess, though not necessary cause NULL bytes are actually catched by `os.Setenv`. It's just the error is ugly and non-explanatory by default (it's just "invalid argument").

I don't quite understand what you're trying to say here, could you rephrase?:-)

Reply:
Basically check for (potential) NULL bytes whenever:

- We set a new manifest
- Someone updates an user-defined secret (which is piped into the environment)

Which should disallow any NULL bytes ideally. If for some other reasons there are still NULL bytes inserted during Marble activation, it will fail anyway, so we do not need any additional checks during runtime in my opinion, except if we want to return a user-friendly error to the user.

Reply:
Thanks! Agreed

Reply:
We now have checking for NULL bytes on manifest set, and every time a Marble activation occurs

For manifest set we do a general check on the env variable string to catch any NULL bytes explicitly written by the person setting the manifest (e.g. a base64 encoded string which contains NULL bytes).

For a Marble activation, this check occurs directly in the template function: If the encoding is of type `string` (formerly `raw`), only data without NULL bytes result in a starting Marble. Otherwise an error is returned notifying the user of the NULL byte.

Reply:
Okay, not I get the runtime checks. But should we not add another check for `WriteSecrets` upon upload of an user-defined secret?

Reply:
I would just re-run the template dry-run. If it's just defined as a file, it should be fine, if it's defined as env, it should fail. Similar to as when you would try to launch such a Marble.

Reply:
Added a template dry run in `WriteSecrets` :)